Framework/Algorithms/src/MergeRuns/SampleLogsBehaviour.cpp

```diff
@@ -0,0 +1,463 @@
+        name + TIME_SERIES_SUFFIX);
+    Kernel::DateAndTime startTime = addeeWS->mutableRun().startTime();
+    double value = addeeWS->mutableRun().getLogAsSingleValue(name);
+    timeSeriesProp->addValue(startTime, value);
+  }
+}
+
+/**
+ * Perform the update for a list property, appending a new value to the existing
+ *string. If the list log entry is in the addeeWS the list log entry is merged
+ *instead.
+ *
+ * @param addeeWS the workspace being merged
+ * @param outWS the workspace the others are merged into
+ * @param addeeWSProperty the relevant property in addeeWS
+ * @param name the name of the property
+ */
+void SampleLogsBehaviour::updateListProperty(
+    const MatrixWorkspace_sptr &addeeWS, const MatrixWorkspace_sptr &outWS,
+    Property *addeeWSProperty, const std::string name) {
+  try {
+    // If this already exists we combine the two strings.
+    auto propertyAddeeWS = addeeWS->getLog(name + LIST_SUFFIX);
+    auto propertyOutWS = outWS->mutableRun().getProperty(name + LIST_SUFFIX);
+    propertyOutWS->setValue(propertyOutWS->value() + ", " +
+                            propertyAddeeWS->value());
+  } catch (std::invalid_argument &) {
+    auto property = outWS->mutableRun().getProperty(name + LIST_SUFFIX);
+    property->setValue(property->value() + ", " + addeeWSProperty->value());
+  }
+}
+
+/**
+ * Performs the check to see if a warning should be generated because logs are
+ *different. Performs a numeric comparison if a tolerance is set and the log is
+ *a number, else performs a string comparison.
+ *
+ * @param addeeWS the workspace being merged
+ * @param addeeWSProperty the property value of the workspace being merged
+ * @param behaviour the merge behaviour struct, containing information about the
+ *merge
+ * @param addeeWSNumber a double for the addeeWS value (0 if it is not numeric)
+ * @param outWSNumber a double for the outWS value (0 if it is not numeric)
+ * @param name the name of the sample log to check
+ */
+void SampleLogsBehaviour::checkWarnProperty(const MatrixWorkspace_sptr &addeeWS,
+                                            Property *addeeWSProperty,
+                                            const SampleLogBehaviour &behaviour,
+                                            const double addeeWSNumber,
+                                            const double outWSNumber,
+                                            const std::string name) {
+  if (behaviour.isNumeric && behaviour.tolerance > 0.0) {
+    if (std::abs(addeeWSNumber - outWSNumber) > behaviour.tolerance) {
+      m_logger.warning() << generateDifferenceMessage(
+          name, addeeWS->name(), addeeWSProperty->value(),
+          behaviour.property->value());
+    }
+  } else {
+    if (behaviour.property->value().compare(addeeWSProperty->value()) != 0) {
+      m_logger.warning() << generateDifferenceMessage(
+          name, addeeWS->name(), addeeWSProperty->value(),
+          behaviour.property->value());
+    }
+  }
```

Comment:
Maybe slightly reorder things such that you do not have to write the call to the logging things twice?